### PR TITLE
FINERACT-1932: Add custom Configuration class for tax service

### PR DESCRIPTION
**Pull request by mmagdiab**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/service/TaxReadPlatformServiceImpl.java [hunk 1]
```diff
@@ -49,7 +48,7 @@ public class TaxReadPlatformServiceImpl implements TaxReadPlatformService {
     private final JdbcTemplate jdbcTemplate;
     private final AccountingDropdownReadPlatformService accountingDropdownReadPlatformService;
 
-    @Autowired
+
     public TaxReadPlatformServiceImpl(final JdbcTemplate jdbcTemplate,
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/service/TaxReadPlatformServiceImpl.java]: Please use Lombok annotations to get rid of the constructor. Let's clean up when we have the opportunity.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxAutoConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@Configuration
+public class TaxAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxAutoConfiguration.java]: Please provide the Java interface class reference in parenthesis with the `@ConditionalOnMissingBean` annotation.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxAutoConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    public TaxReadPlatformService taxReadPlatformService(JdbcTemplate jdbcTemplate,
+            AccountingDropdownReadPlatformService accountingDropdownReadPlatformService) {
+        return new TaxReadPlatformServiceImpl(jdbcTemplate, accountingDropdownReadPlatformService);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
```
Comment (vidakovic) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxAutoConfiguration.java]: Please provide the Java interface class reference in parenthesis with the `@ConditionalOnMissingBean` annotation.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/service/TaxReadPlatformServiceImpl.java (hunk 1 above)
Comment (mmagdiab) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/service/TaxReadPlatformServiceImpl.java]: done, updated.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxAutoConfiguration.java (hunk 3 above)
Comment (mmagdiab) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxAutoConfiguration.java]: done, Updated

**Review by adamsaghy — APPROVED** [review]
LGTM

**Review by galovics — APPROVED** [review]

**Comment by adamsaghy** [comment]
@mmagdiab Please kindly check the below error:
```
    (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
  Did you mean 'StringUtils.capitalize(accountType.toString().toLowerCase(Locale.ROOT)) + " account with identifier " + id + " does not exist", id, e);' or 'StringUtils.capitalize(accountType.toString().toLowerCase(Locale.getDefault())) + " account with identifier " + id + " does not exist", id, e);' or 'StringUtils.capitalize(Ascii.toLowerCase(accountType.toString())) + " account with identifier " + id + " does not exist", id, e);'?
/home/runner/work/fineract/fineract/fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxConfiguration.java:29: error: [WildcardImport] Wildcard imports, static or otherwise, should not be used
import org.apache.fineract.portfolio.tax.service.*;
```

**Comment by mmagdiab** [comment]
> @mmagdiab Please kindly check the below error:
> 
> ```
>     (see https://errorprone.info/bugpattern/StringCaseLocaleUsage)
>   Did you mean 'StringUtils.capitalize(accountType.toString().toLowerCase(Locale.ROOT)) + " account with identifier " + id + " does not exist", id, e);' or 'StringUtils.capitalize(accountType.toString().toLowerCase(Locale.getDefault())) + " account with identifier " + id + " does not exist", id, e);' or 'StringUtils.capitalize(Ascii.toLowerCase(accountType.toString())) + " account with identifier " + id + " does not exist", id, e);'?
> /home/runner/work/fineract/fineract/fineract-provider/src/main/java/org/apache/fineract/portfolio/tax/starter/TaxConfiguration.java:29: error: [WildcardImport] Wildcard imports, static or otherwise, should not be used
> import org.apache.fineract.portfolio.tax.service.*;
> ```

resolved

**Pull request merged**